common/src/point_cloud_utils.cpp

```diff
@@ -80,14 +88,67 @@ namespace mas_perception_libs
     {
         PointCloud::Ptr filteredCloudPtr = boost::make_shared<PointCloud>();
 
-        mPassThroughFilterX.setInputCloud(pCloudPtr);
-        mPassThroughFilterX.filter(*filteredCloudPtr);
+         if (isPassthroughFilterEnabled_)
+        {
+            if (isPassthroughFilterXEnabled_ && isPassthroughFilterYEnabled_ && isPassthroughFilterZEnabled_)
```

Comment:
Instead of an exhaustive nested if, why don't you simply use three individual if statements here, namely

```
if (isPassthroughFilterXEnabled_)
{
    mPassThroughFilterX.setInputCloud(pCloudPtr);
    mPassThroughFilterX.filter(*filteredCloudPtr);
}

if (isPassthroughFilterYEnabled_)
{
    mPassThroughFilterY.setInputCloud(filteredCloudPtr);
    mPassThroughFilterY.filter(*filteredCloudPtr);
}

if (isPassthroughFilterZEnabled_)
{
    mPassThroughFilterZ.setInputCloud(filteredCloudPtr);
    mPassThroughFilterZ.filter(*filteredCloudPtr);
}
```

Reply:
the problem is, we may want to enable Y or Z only, so your code does not work on this condition since the source now is still based on filteredCloudPtr (which is an empty cloud since passtrhoughX is not executed)

Reply:
True, true. `pCloudPtr` should be the input cloud in all three conditions.

Reply:
But we cannot make it like that because we may also want to enable two or three filters. Let's say we want to enable X and Y, the input of X is pCloudPtr, but the input of Y now is filteredCloudPtr (which is the filteredCloud from X)

Reply:
Or is the code relying on side effects of `mPassThroughFilter<K>` and my version sets an incorrect result to `filteredCloudPtr`?

Reply:
So side effects it is.

Reply:
Yes

Reply:
copy `pCloudPtr` into `filteredCloudPtr`, then go through the 3 if statements as Alex wrote with the input to all filters as `filteredCloudPtr`. Does that work?

Reply:
how about:
```cpp
PointCloud::Ptr filteredCloudPtr = nullptr;

if (isPassthroughFilterXEnabled_) {
    if (filteredCloudPtr == nullptr) {
        filteredCloudPtr = boost::make_shared<PointCloud>();
    }
    mPassThroughFilterX.setInputCloud(pCloudPtr);
    mPassThroughFilterX.filter(*filteredCloudPtr);
}
```
multiple `make_shared` but less ugly if's

Reply:
and also no need for a global `isPassthroughFilterXEnabled`

Reply:
actually @sthoduka's way is probably the cleanest. Anyway is fine for me. Just make sure to run tests afterwards.